server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerCassandraRule.java

```diff
@@ -0,0 +1,88 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+import java.util.Arrays;
+
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.james.modules.mailbox.CassandraSessionConfiguration;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.shaded.com.github.dockerjava.api.model.ExposedPort;
+import org.testcontainers.shaded.com.github.dockerjava.api.model.Ports;
+
+import com.google.inject.Module;
+
+
+public class DockerCassandraRule implements GuiceModuleTestRule {
+
+    private static final int CASSANDRA_PORT = 9042;
+    private GenericContainer<?> cassandraContainer = new GenericContainer<>("cassandra:2.2");
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return cassandraContainer.apply(base, description);
+    }
+
+    @Override
+    public void await() {
+    }
+
+    @Override
+    public Module getModule() {
+        return (binder) -> binder.bind(CassandraSessionConfiguration.class).toInstance(this::getCassandraConfigurationForDocker);
+    }
+
+    @SuppressWarnings("deprecation")
+    public String getIp() {
+        return cassandraContainer.getContainerInfo().getNetworkSettings().getIpAddress();
+    }
+
+    public int getBindingPort() {
+        Ports.Binding[] bindings =  cassandraContainer
+                .getContainerInfo()
+                .getNetworkSettings()
+                .getPorts()
+                .getBindings()
+                .get(ExposedPort.tcp(CASSANDRA_PORT));
+
+        return Integer.parseInt(
```

Comment:
Ok but why reading that : http://stackoverflow.com/questions/7355024/integer-valueof-vs-integer-parseint. In this case I can't see good reason to prefer valueOf